src/mscorlib/src/System/Reflection/Assembly.cs

```diff
@@ -335,7 +351,7 @@ public static Assembly Load(AssemblyName assemblyRef)
 #endif
 
             StackCrawlMark stackMark = StackCrawlMark.LookForMyCaller;
```

Comment:
I think the stackmark should stay in the public method and be passed in as argument to the internal helper method. Otherwise, the stackwalk will identify System.Private.CoreLib as the caller.